packages/subgraph/src/utils.ts

```diff
@@ -106,4 +106,116 @@ export function decodePermissions(permissions: readonly string[]): Permission[]
     return decodedPermissions
 }
 
+export async function updateSpaceCachedMetrics(
+    context: Context,
+    spaceId: `0x${string}`,
+): Promise<void> {
+    // Get current timestamp for rolling window calculations
+    const currentTimestamp = Math.floor(Date.now() / 1000)
+    const oneDayAgo = currentTimestamp - 86400
+    const sevenDaysAgo = currentTimestamp - 7 * 86400
+    const thirtyDaysAgo = currentTimestamp - 30 * 86400
+
+    // Get current space
+    const space = await context.db.sql.query.space.findFirst({
+        where: eq(schema.space.id, spaceId),
+    })
+
+    if (!space) {
+        console.warn(`Space ${spaceId} not found`)
+        return
+    }
+
+    // Query ALL events for all-time totals
+    const allEvents = await context.db.sql.query.analyticsEvent.findMany({
+        where: eq(schema.analyticsEvent.spaceId, spaceId),
+    })
+
+    // Calculate all metrics from scratch
+    let swapVolume24h = 0n
+    let swapVolume7d = 0n
+    let swapVolume30d = 0n
+    let swapVolume = 0n
+    let tipVolume24h = 0n
+    let tipVolume7d = 0n
+    let tipVolume30d = 0n
+    let tipVolume = 0n
+    let joinVolume24h = 0n
+    let joinVolume7d = 0n
+    let joinVolume30d = 0n
+    let joinVolume = 0n
+    let memberCount24h = 0n
+    let memberCount7d = 0n
+    let memberCount30d = 0n
+    let memberCount = 0n
+
+    for (const event of allEvents) {
```

Comment:
I believe the api we want here is for this side effect to kick off on the event type that has fired in the analytic table, ie tip, swap, etc, rather than recalculating all metrics for every event that is added. Can we update this function to take an event type and only update metrics for spaceId, eventType ?